src/mqt/problemsolver/partialcompiler/Partial_QAOA.py

```diff
@@ -0,0 +1,192 @@
+        qc_prep.h(range(self.num_qubits))
+        qc_prep.barrier()
+        qcs_problem = []
+        qcs_mix = []
+        self.problem_parameters = []
+        for i in range(self.repetitions):
+            qc_problem = QuantumCircuit(self.num_qubits)
+            p = Parameter(f"a_{i}")
+            self.problem_parameters.append(p)
+            for elem in self.offline_time_edges:
+                qc_problem.rzz(p, elem[0], elem[1])
+            if include_online_edges:
+                for elem in self.online_time_edges:
+                    qc_problem.rzz(p, elem[0], elem[1])
+
+            qcs_problem.append(qc_problem)
+
+            m = Parameter(f"b_{i}")
+            qc_mix = QuantumCircuit(self.num_qubits)
+            qc_mix.barrier()
+            qc_mix.rx(2 * m, range(self.num_qubits))
+            qc_mix.barrier()
+            qcs_mix.append(qc_mix)
+
+        return qc_prep, qcs_problem, qcs_mix
+
+    def determine_mapping(self) -> list[int]:
+        """Determines a mapping layout to the selected quantum devices based on the offline edges."""
+        qc_prep, qcs_problem_uncompiled, qcs_mix_uncompiled = self.get_uncompiled_circuit()
+        assert len(qcs_problem_uncompiled) == len(qcs_mix_uncompiled)
+        qc_composed = qc_prep.copy()
+        for i in range(len(qcs_problem_uncompiled)):
+            qc_composed.compose(qcs_problem_uncompiled[i], inplace=True)
+            qc_composed.compose(qcs_mix_uncompiled[i], inplace=True)
+
+        offline_mapped_qc = transpile(
+            qc_composed, coupling_map=self.coupling_map, basis_gates=IBM_GATES, optimization_level=3
+        )
+
+        layout = offline_mapped_qc._layout.initial_layout
+        mapping = []
+        for elem in layout.get_virtual_bits():
+            if elem.register.name == "ancilla":
+                pass
+            else:
+                mapping.append(layout.get_virtual_bits()[elem])
+        return mapping
+
+    def get_partially_compiled_circuit_without_online_edges(
+        self, consider_mapping: bool = True
+    ) -> tuple[QuantumCircuit, list[QuantumCircuit], list[QuantumCircuit]]:
+        """
+        Return the state preparation circuit and lists of problem and mixer circuits with the option to consider
+        the mapping with all returned circuits are compiled to native gates and optionally mapped to the device.
+        """
+
+        self.mapping = self.determine_mapping()
+
+        qc_prep, qcs_problem_uncompiled, qcs_mix_uncompiled = self.get_uncompiled_circuit()
+
+        mapping_fct = self.compile_with_mapping if consider_mapping else self.compile_without_mapping
+        qc_prep_compiled = mapping_fct(qc_prep)
+        qcs_problem_compiled = []
+        qcs_mixer_compiled = []
+        for i in range(len(qcs_problem_uncompiled)):
+            qcs_problem_compiled.append(mapping_fct(qcs_problem_uncompiled[i]))
+            qcs_mixer_compiled.append(mapping_fct(qcs_mix_uncompiled[i]))
```

Comment:
you could just write these as generator expressions. Might be more efficient and pythonic.
(If you use GitHub CoPilot, it should even suggest it to you if you delete the code and type
```
qcs_problem_compiled = [
```

Reply:
Done.